packages/block-library/src/image/index.php

```diff
@@ -31,68 +31,90 @@ function render_block_core_image( $attributes, $content, $block ) {
+			$lightbox_enabled = true;
+		}
 	}
 
 	// If at least one block in the page has the lightbox, mark the block type as interactive.
-	if ( $should_load_view_script ) {
+	if ( $lightbox_enabled ) {
 		$block->block_type->supports['interactivity'] = true;
 	}
 
+	// Determine whether the view script should be enqueued or not.
 	$view_js_file = 'wp-block-image-view';
 	if ( ! wp_script_is( $view_js_file ) ) {
 		$script_handles = $block->block_type->view_script_handles;
 
 		// If the script is not needed, and it is still in the `view_script_handles`, remove it.
-		if ( ! $should_load_view_script && in_array( $view_js_file, $script_handles, true ) ) {
+		if ( ! $lightbox_enabled && in_array( $view_js_file, $script_handles, true ) ) {
 			$block->block_type->view_script_handles = array_diff( $script_handles, array( $view_js_file ) );
 		}
 		// If the script is needed, but it was previously removed, add it again.
-		if ( $should_load_view_script && ! in_array( $view_js_file, $script_handles, true ) ) {
+		if ( $lightbox_enabled && ! in_array( $view_js_file, $script_handles, true ) ) {
 			$block->block_type->view_script_handles = array_merge( $script_handles, array( $view_js_file ) );
 		}
 	}
 
+	if ( $lightbox_enabled ) {
+		return block_core_image_render_lightbox( $processor->get_updated_html(), $block->parsed_block );
+	}
+
 	return $processor->get_updated_html();
 }
 
-
 /**
- * Add the directives and layout needed for the lightbox behavior.
+ * Add the lightboxEnabled flag to the block data.
  *
- * @param  string $block_content Rendered block content.
- * @param  array  $block         Block object.
- * @return string                Filtered block content.
+ * This is used to determine whether the lightbox should be rendered or not.
+ *
+ * @param  array $block Block data.
+ * @return array        Filtered block data.
  */
-function block_core_image_render_lightbox( $block_content, $block ) {
-	$link_destination = isset( $block['attrs']['linkDestination'] ) ? $block['attrs']['linkDestination'] : 'none';
-
+function block_core_image_get_lightbox_settings( $block ) {
 	// Get the lightbox setting from the block attributes.
 	if ( isset( $block['attrs']['lightbox'] ) ) {
 		$lightbox_settings = $block['attrs']['lightbox'];
+		// If the lightbox setting is not set in the block attributes,
+		// check the legacy lightbox settings that are set using the
+		// `gutenberg_should_render_lightbox` filter.
+		// We can remove this elseif statement when the legacy lightbox settings are removed.
+	} elseif ( isset( $block['legacyLightboxSettings'] ) ) {
+		$lightbox_settings = $block['legacyLightboxSettings'];
 	}
 
-	if ( ! isset( $lightbox_settings ) || 'none' !== $link_destination ) {
-		return $block_content;
+	if ( ! isset( $lightbox_settings ) ) {
+		$lightbox_settings = gutenberg_get_global_settings( array( 'lightbox' ), array( 'block_name' => 'core/image' ) );
```

Comment:
We can't use this function here because block PHP files are auto-generated in core from the package. Any `gutenberg` namespaced function is exclusive to the plugin so will error in core.

I thought we'd added a check for gutenberg namespaces in block-library? Cc @anton-vlasenko and @gziolo any idea how this might have slipped through?

Reply:
The check for function names at the moment covers only definitions. Preventing the usage of functions prefixed with `gutenberg_` could be the next step. The challenge though is that we still need to allow them when wrapped with:

```php
if ( defined( 'IS_GUTENBERG_PLUGIN' ) && IS_GUTENBERG_PLUGIN ) {}
```

I know it's possible with ESLint for JavaScript, but I'm not sure about linting for PHP.

Reply:
Oooh I see, thanks for explaining!

Reply:
cc @anton-vlasenko

Reply:
Thank you for the ping, @Mamaduka and @tellthemachines. I've created a new GitHub issue for this and hope to submit a PR soon: https://github.com/WordPress/gutenberg/issues/54745